### PR TITLE
EVO color customization

### PR DESCRIPTION
**Pull request by YanningTan**
Implemented international palettes and allowed customization for EVO

### CODE REVIEW

File: leaf-ui/analysis.html [hunk 1]
```diff
@@ -57,7 +57,8 @@
                 <!-- slider div -->
                 <input type="range" min="0" max="2" value="0" class="EVO-slider" id="colorResetAnalysis">
                 <!-- makes slider -->
-                <p id="top-label-analysis" style="display: inside">EVO</p> <!-- sat slider label -->
+                <p id="top-label-analysis" style="display: inside">EVO</p>
+                <!-- sat slider label -->
```
Comment (YesugenB) [inline leaf-ui/analysis.html:61]: Does this label apply to line 60 or 62? Why move it

File: leaf-ui/index.html [hunk 2]
```diff
@@ -74,44 +74,114 @@
...
         <button id="btn-view-assignment" class="btn model-only" style="position:relative; border-radius:none;">View
             Assignments List</button>
-        <!--This is the modal box of assignments-->
+        
+        <!--This is the modal box of Intermediate Values-->
         <button id="btn-view-intermediate" class="btn model-only">Intermediate
             Values</button>
-        <!--This is the modal box of Intermediate Values-->
-
+
```
Comment (YesugenB) [inline leaf-ui/index.html:122]: Extra line?

File: leaf-ui/js/object/evoObjects.js [hunk 3]
```diff
@@ -25,19 +25,92 @@ class IntentionColorVis {
...
  */
 class EVO {
+
     static colorVisDict = {
+        "0000": "#D3D3D3", // None
+        "0011": "#003fff", // Satisfied
+        "0010": "#8FB8DE", // Partially satisfied
+        "0100": "#fbaca8", // Partially denied
+        "0110": "#9400D3", // Conflict 
+        "0111": "#5946b2",
```
Comment (YesugenB) [inline leaf-ui/js/object/evoObjects.js]: I think this is a conflict right? Should we add a comment for it, and in lines 40 and 41

File: leaf-ui/js/object/evoObjects.js [hunk 4]
```diff
@@ -461,12 +526,31 @@ class EVO {
...
+        }
+
+        if (EVO.paletteOption == 6) {
+            EVO.fillInDictionary();
+            return EVO.selfColorVisDict[intentionEval];
         }
-        return EVO.colorVisDict[intentionEval];
     }
 
+    static validateColor(color) {
```
Comment (YesugenB) [inline leaf-ui/js/object/evoObjects.js]: Add a docstring for this function?

File: leaf-ui/js/object/evoObjects.js [hunk 5]
```diff
@@ -631,7 +739,20 @@ class EVONextState {
...
+        }
+        if (EVONextState.paletteOption == 5) {
+            var selfVis = myInputJSObject.results.get('colorVis').selfColorVisDict;
+
+            return selfVis[intentionEval];
+        }
+
+    }
+
+        //return EVO.colorVisDict[intentionEval]; }
```
Comment (YesugenB) [inline leaf-ui/js/object/evoObjects.js]: Are you saving this line for something later?

**Review by YesugenB — CHANGES_REQUESTED** [review]
Looks good! 😎 Just noticed a few things and had some questions

- The EVO toolbar is longer in analysis mode and the spacing doesn't line up with the text
- For `My Palette` just opening the pop-up and closing it without making any changes gives the `Invalid Color` swal error, and turns some of the intentions fully black
Screen Shot 2022-04-21 at 4 30 18 PM

- For the yellow-purple color palette, maybe change the text color or the shade of the yellows. Once simulate single path has been run, it's hard to see the text. Also the lime green do be harsh on the eyes
- When your EVO is in `state` mode in Analysis Mode, and you change color palettes, the slider disappears and in general, the new palette doesn't automatically show up - you have to go to a different state and go back to the one you were on

File: leaf-ui/analysis.html (hunk 1 above)
Comment (YanningTan) [reply leaf-ui/analysis.html:61]: line 60, just to make it the same format as other docstrings

File: leaf-ui/js/object/evoObjects.js (hunk 3 above)
Comment (YanningTan) [reply leaf-ui/js/object/evoObjects.js]: great idea! now fully documented

File: leaf-ui/js/object/evoObjects.js (hunk 5 above)
Comment (YanningTan) [reply leaf-ui/js/object/evoObjects.js]: just removed!

**Comment by YanningTan** [comment]
> Looks good! 😎 Just noticed a few things and had some questions
> 
> * The EVO toolbar is longer in analysis mode and the spacing doesn't line up with the text
> * For `My Palette` just opening the pop-up and closing it without making any changes gives the `Invalid Color` swal error, and turns some of the intentions fully black
> 
> Screen Shot 2022-04-21 at 4 30 18 PM
> 
> * For the yellow-purple color palette, maybe change the text color or the shade of the yellows. Once simulate single path has been run, it's hard to see the text. Also the lime green do be harsh on the eyes
> * When your EVO is in `state` mode in Analysis Mode, and you change color palettes, the slider disappears and in general, the new palette doesn't automatically show up - you have to go to a different state and go back to the one you were on



> Looks good! 😎 Just noticed a few things and had some questions
> 
> * The EVO toolbar is longer in analysis mode and the spacing doesn't line up with the text
> * For `My Palette` just opening the pop-up and closing it without making any changes gives the `Invalid Color` swal error, and turns some of the intentions fully black
> 
> Screen Shot 2022-04-21 at 4 30 18 PM
> 
> * For the yellow-purple color palette, maybe change the text color or the shade of the yellows. Once simulate single path has been run, it's hard to see the text. Also the lime green do be harsh on the eyes
> * When your EVO is in `state` mode in Analysis Mode, and you change color palettes, the slider disappears and in general, the new palette doesn't automatically show up - you have to go to a different state and go back to the one you were on

Thanks a lot! The first and the last problem do not appear on my end🤔. The second one is fixed!

**Comment by YanningTan** [comment]
> Looks good! 😎 Just noticed a few things and had some questions
> 
> * The EVO toolbar is longer in analysis mode and the spacing doesn't line up with the text
> * For `My Palette` just opening the pop-up and closing it without making any changes gives the `Invalid Color` swal error, and turns some of the intentions fully black
> 
> Screen Shot 2022-04-21 at 4 30 18 PM
> 
> * For the yellow-purple color palette, maybe change the text color or the shade of the yellows. Once simulate single path has been run, it's hard to see the text. Also the lime green do be harsh on the eyes
> * When your EVO is in `state` mode in Analysis Mode, and you change color palettes, the slider disappears and in general, the new palette doesn't automatically show up - you have to go to a different state and go back to the one you were on
Thanks a lot! The first and the last problem do not appear on my end🤔. The second one is fixed!

**Comment by YanningTan** [comment]
> Looks good! 😎 Just noticed a few things and had some questions
> 
> * The EVO toolbar is longer in analysis mode and the spacing doesn't line up with the text
> * For `My Palette` just opening the pop-up and closing it without making any changes gives the `Invalid Color` swal error, and turns some of the intentions fully black
> 
> Screen Shot 2022-04-21 at 4 30 18 PM
> 
> * For the yellow-purple color palette, maybe change the text color or the shade of the yellows. Once simulate single path has been run, it's hard to see the text. Also the lime green do be harsh on the eyes
> * When your EVO is in `state` mode in Analysis Mode, and you change color palettes, the slider disappears and in general, the new palette doesn't automatically show up - you have to go to a different state and go back to the one you were on



Thanks a lot! The first and the last problem do not appear on my end🤔. The second one is fixed!

**Comment by YesugenB** [comment]
@YanningTan The slider looks like this one mine! I'm on a mac maybe that's why? 
Screen Shot 2022-04-25 at 9 50 31 AM
Screen Shot 2022-04-25 at 9 51 38 AM

**Review by YesugenB — APPROVED** [review]
Looks good! I made a video of the slider disappearing when you change color palettes, but everything else is great. Maybe make an issue for it?

https://user-images.githubusercontent.com/59470488/165840701-49d9d8ca-af47-4f47-98c6-098fdbf126e6.mov

**Pull request merged**